libsel4vmmplatsupport/include/sel4vmmplatsupport/drivers/virtio_blk.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2020, Data61
+ * Commonwealth Scientific and Industrial Research Organisation (CSIRO)
+ * ABN 41 687 119 230.
+ *
+ * This software may be distributed and modified according to the terms of
+ * the BSD 2-Clause license. Note that NO WARRANTY is provided.
+ * See "LICENSE_BSD2.txt" for details.
+ *
+ * @TAG(DATA61_BSD)
+ */
+
+#pragma once
+
+#include <sel4vm/guest_vm.h>
+
+#include <sel4vmmplatsupport/ioports.h>
+#include <sel4vmmplatsupport/drivers/pci.h>
+#include <sel4vmmplatsupport/drivers/virtio_pci_emul.h>
+
+typedef struct virtio_blk {
+    unsigned int iobase;
+    virtio_emul_t *emul;
+    struct disk_driver *emul_driver;
+    raw_diskiface_funcs_t emul_driver_funcs;
+    ps_io_ops_t ioops;
+} virtio_blk_t;
+
+/* TODO Check if emulate_bar_access is needed */
+virtio_blk_t *common_make_virtio_blk(vm_t *vm, vmm_pci_space_t *pci, vmm_io_port_list_t *ioport,
+                                     ioport_range_t ioport_range, ioport_type_t port_type,
```

Comment:
This seems like odd formatting

Reply:
Not quite sure what you mean here.

Reply:
Oh weird, must have been my browser rendering.